homeassistant/components/vallox/sensor.py

```diff
@@ -264,16 +119,126 @@ class ValloxFilterRemainingSensor(ValloxSensor):
     async def async_update(self):
         """Fetch state from the ventilation unit."""
         try:
-            days_remaining = int(self._state_proxy.fetch_metric(self._metric_key))
+            days_remaining = int(
+                self._state_proxy.fetch_metric(self.entity_description.metric_key)
+            )
             days_remaining_delta = timedelta(days=days_remaining)
 
             # Since only a delta of days is received from the device, fix the
             # time so the timestamp does not change with every update.
             now = datetime.utcnow().replace(hour=13, minute=0, second=0, microsecond=0)
 
-            self._state = (now + days_remaining_delta).isoformat()
-            self._available = True
+            self._attr_native_value = (now + days_remaining_delta).isoformat()
+            self._attr_available = True
 
         except (OSError, KeyError) as err:
-            self._available = False
+            self._attr_available = False
             _LOGGER.error("Error updating sensor: %s", err)
+
+
+@dataclass
+class ValloxSensorEntityDescription(SensorEntityDescription):
+    """Describes Vallox sensor entity."""
+
+    metric_key: str | None = None
+    sensor_type: type[ValloxSensor] = ValloxSensor
+
+
+SENSORS: tuple[ValloxSensorEntityDescription, ...] = (
+    ValloxSensorEntityDescription(
+        key="current_profile",
+        name="Current Profile",
+        icon="mdi:gauge",
+        sensor_type=ValloxProfileSensor,
+    ),
+    ValloxSensorEntityDescription(
+        key="fan_speed",
+        name="Fan Speed",
+        metric_key="A_CYC_FAN_SPEED",
+        icon="mdi:fan",
+        state_class=STATE_CLASS_MEASUREMENT,
+        native_unit_of_measurement=PERCENTAGE,
+        sensor_type=ValloxFanSpeedSensor,
+    ),
+    ValloxSensorEntityDescription(
+        key="remaining_time_for_filter",
+        name="Remaining Time For Filter",
+        metric_key="A_CYC_REMAINING_TIME_FOR_FILTER",
+        icon="mdi:filter",
```

Comment:
```suggestion
```
We don't add icons to entities with a device class.

Reply:
Hmm, this would make this patch not a pure refactor anymore, since this sensor had the filter icon since the very beginning?

I understand the guideline, but at the same time, the filter icon makes a lot of sense for this special sensor, doesn’t it?

I’m good to change it if getting the patch merged depends on it? 👍

Reply:
I don't think it does, it's a timestamp sensor, the same as any other, we should be consistent :+1: